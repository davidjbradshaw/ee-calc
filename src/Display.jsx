import React from 'react'
import PropTypes from 'prop-types'

import { eeBlue } from './consts'

const displayStyle={
  border: `solid 5px ${eeBlue}`,
  color: eeBlue,
  backgroundColor: 'white',
  borderRadius: '5px',
  fontSize: '20px',
  padding: `5px 10px 5px`,
  textAlign: 'right',
  overflow: 'hidden',
}

const Display = ({children}) => (
  <div style={displayStyle}>{ children }</div>
)

Display.prototypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
}

export default Display