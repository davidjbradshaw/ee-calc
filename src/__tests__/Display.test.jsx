import React from 'react'
import { shallow } from 'enzyme'

import Display from '../Display'

it('Renders the display', () => {
  const value = 123
  const display = <Display>{value}</Display>

  expect(display).toMatchSnapshot()

  const component = shallow(display)

  expect(component.text()).toBe(value.toString())
})
