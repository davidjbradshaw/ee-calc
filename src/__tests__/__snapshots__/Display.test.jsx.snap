// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the display 1`] = `
<Display>
  123
</Display>
`;
